src/test/hosting/api.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { expect } from "chai";
+import * as nock from "nock";
+import * as sinon from "sinon";
+import * as helpers from "../helpers";
+import * as api from "../../api";
+import * as hostingApi from "../../hosting/api";
+
+const TEST_CHANNELS_RESPONSE = {
+  channels: [
+    // domain exists in TEST_GET_DOMAINS_RESPONSE
+    { url: "https://my-site--ch1-4iyrl1uo.web.app" },
+    // domain does not exist in TEST_GET_DOMAINS_RESPONSE
+    // we assume this domain was manually removed by
+    // the user from the identity api
+    { url: "https://my-site--ch2-ygd8582v.web.app" },
+  ],
+};
+const TEST_GET_DOMAINS_RESPONSE = {
+  authorizedDomains: [
+    "my-site.firebaseapp.com",
+    "localhost",
+    "randomurl.com",
+    "my-site--ch1-4iyrl1uo.web.app",
+    // domain that should be removed
+    "my-site--expiredchannel-difhyc76.web.app",
+  ],
+};
+
+const EXPECTED_DOMAINS_RESPONSE = [
+  "my-site.firebaseapp.com",
+  "localhost",
+  "randomurl.com",
+  "my-site--ch1-4iyrl1uo.web.app",
+];
+const PROJECT_ID = "test-project";
+const SITE = "my-site";
+
+describe("hosting", () => {
+  beforeEach(() => {
+    helpers.mockAuth(sinon);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("getApprovedDomains", () => {
+    afterEach(() => {
+      nock.cleanAll();
+    });
+
+    it("it should return the list of expected auth domains after syncing", async () => {
```

Comment:
the "it" doesn't need to be repeated

```suggestion
    it("should return the list of expected auth domains after syncing", async () => {
```